sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2Suite.scala

```diff
@@ -396,87 +392,66 @@ object SimpleReaderFactory extends PartitionReaderFactory {
   }
 }
 
-abstract class SimpleReadSupport extends BatchReadSupport {
-  override def fullSchema(): StructType = new StructType().add("i", "int").add("j", "int")
-
-  override def newScanConfigBuilder(): ScanConfigBuilder = {
-    NoopScanConfigBuilder(fullSchema())
-  }
+abstract class SimpleBatchTable extends Table with SupportsBatchRead  {
 
-  override def createReaderFactory(config: ScanConfig): PartitionReaderFactory = {
-    SimpleReaderFactory
-  }
+  override def schema(): StructType = new StructType().add("i", "int").add("j", "int")
 }
 
+abstract class SimpleScanBuilder extends ScanBuilder
+  with Batch with Scan {
```

Comment:
I like that the API is flexible enough that `ScanBuilder`, `Scan`, and `Batch` can be the same object in simple cases.